apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerDbTypeTests.ValueConversions.cs

```diff
@@ -535,30 +557,64 @@ public static IEnumerable<object[]> GetValidValueConversions()
-                Value.ForString(Convert.ToBase64String(Duration.FromTimeSpan(TimeSpan.FromSeconds(10)).ToByteArray())),
-                SpannerDbType.FromClrType(typeof(Duration)),
-                typeof(Duration)
-            },
+                new List<Rectangle>{ testRectangle1, testRectangle2 },
+                SpannerDbType.ArrayOf(SpannerDbType.FromClrType(typeof(Rectangle))),
+                $"[ {testRectangle1Wire}, {testRectangle2Wire} ]",
+            };
+
+            yield return new object[]
+                {   // We perform automatic serialization/deserialization.
+                testPerson1,
+                SpannerDbType.FromClrType(typeof(Person)),
+                testPerson1Wire.ToString(),
+                };
+            yield return new object[]
             {   // The value is serialized/deserialized by calling code.
-                Value.ForString(Convert.ToBase64String(Duration.FromTimeSpan(TimeSpan.FromSeconds(10)).ToByteArray())),
-                Value.ForString(Convert.ToBase64String(Duration.FromTimeSpan(TimeSpan.FromSeconds(10)).ToByteArray())),
-                SpannerDbType.ForProtobuf(Duration.Descriptor.FullName),
-                typeof(Value)
-            },
-            {   // Value is being itself used as a type.
-                Value.ForBool(true),
-                Value.ForString(Convert.ToBase64String(Value.ForBool(true).ToByteArray())),
-                SpannerDbType.FromClrType(typeof(Value)),
-                typeof(Value)
-            }
-        };
+                testPerson1,
+                SpannerDbType.ForProtobuf(Person.Descriptor.FullName),
+                testPerson1Wire.ToString(),
+            };
+            yield return new object[]
+            {   // We perform automatic serialization/deserialization.
+                new List<Person>{ testPerson1, testPerson2 },
+                SpannerDbType.ArrayOf(SpannerDbType.FromClrType(typeof(Person))),
+                $"[ {testPerson1Wire}, {testPerson2Wire} ]",
+            };
+
+            yield return new object[]
+                {   // We perform automatic serialization/deserialization.
+                testValueWrapper1,
+                SpannerDbType.FromClrType(typeof(ValueWrapper)),
+                testValueWrapper1Wire.ToString(),
+                };
+            yield return new object[]
+            {   // The value is serialized/deserialized by calling code.
+                testValueWrapper1,
+                SpannerDbType.ForProtobuf(ValueWrapper.Descriptor.FullName),
+                testValueWrapper1Wire.ToString(),
+            };
+            yield return new object[]
+            {   // We perform automatic serialization/deserialization.
+                new List<ValueWrapper>{ testValueWrapper1, testValueWrapper2 },
+                SpannerDbType.ArrayOf(SpannerDbType.FromClrType(typeof(ValueWrapper))),
+                $"[ {testValueWrapper1Wire}, {testValueWrapper2Wire} ]",
```

Comment:
So this is JSON, but of base64 protobuf binary wire format values, right?

Reply:
Yes, exactly.